FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1457 @@
+            break;
+
+        case PUBLISH:
+            assert( pxCommand->pContext != NULL );
+            pxPublishInfo = pxCommand->pContext->pPublishInfo;
+            assert( pxPublishInfo != NULL );
+
+            if( pxPublishInfo->qos != MQTTQoS0 )
+            {
+                usPacketId = MQTT_GetPacketId( &globalMqttContext );
+            }
+
+            LogInfo( ( "Publishing message to %.*s.", ( int ) pxPublishInfo->topicNameLength, pxPublishInfo->pTopicName ) );
+            xStatus = MQTT_Publish( &globalMqttContext, pxPublishInfo, usPacketId );
+            pxCommand->pContext->returnStatus = xStatus;
+
+            /* Add to pending ack list, or call callback if QoS 0. */
+            xAddAckToList = ( pxPublishInfo->qos != MQTTQoS0 ) && ( xStatus == MQTTSuccess );
+            break;
+
+        /* TODO: Option to subscribe/unsubscribe without sending a packet,
+         * e.g. for Shadow topics. */
+        case SUBSCRIBE:
+        case UNSUBSCRIBE:
+            assert( pxCommand->pContext != NULL );
+            pxSubscribeInfo = pxCommand->pContext->pSubscribeInfo;
+            assert( pxSubscribeInfo != NULL );
+            assert( pxSubscribeInfo->pTopicFilter != NULL );
+            usPacketId = MQTT_GetPacketId( &globalMqttContext );
+
+            if( pxCommand->commandType == SUBSCRIBE )
+            {
+                LogInfo( ( "Subscribing to %.*s",
+                           pxSubscribeInfo->topicFilterLength,
+                           pxSubscribeInfo->pTopicFilter ) );
+                xStatus = MQTT_Subscribe( &globalMqttContext,
+                                          pxSubscribeInfo,
+                                          pxCommand->pContext->subscriptionCount,
+                                          usPacketId );
+            }
+            else
+            {
+                LogInfo( ( "Unsubscribing from %.*s", pxSubscribeInfo->topicFilterLength, pxSubscribeInfo->pTopicFilter ) );
+                xStatus = MQTT_Unsubscribe( &globalMqttContext,
+                                            pxSubscribeInfo,
+                                            pxCommand->pContext->subscriptionCount,
+                                            usPacketId );
+            }
+
+            pxCommand->pContext->returnStatus = xStatus;
+            xAddAckToList = ( xStatus == MQTTSuccess );
+            break;
+
+        case PING:
+            xStatus = MQTT_Ping( &globalMqttContext );
+
+            if( pxCommand->pContext != NULL )
+            {
+                pxCommand->pContext->returnStatus = xStatus;
+            }
+
+            break;
+
+        case DISCONNECT:
+            xStatus = MQTT_Disconnect( &globalMqttContext );
+
+            if( pxCommand->pContext != NULL )
+            {
+                pxCommand->pContext->returnStatus = xStatus;
+            }
+
+            break;
+
+        case CONNECT:
+            /* TODO: Reconnect. */
```

Comment:
Is this TODO planning to be addressed as part of this PR?

Reply:
I was planning to make it a separate PR, since adding more functions for reconnect would make this PR larger than I would like

Reply:
Sounds good. Thanks